### ISSUE TITLE
write txt files containing dipoles and spiking

### ISSUE THREAD

**Issue by rythorpe** [jonescompneurolab/hnn-core#95]
It looks like a dipole.write() function exists but it needs to be fixed so that the name is assigned appropriately for each dipole file created.

A similar function would be great for network spiking so that we can generate data files similar to those produced by HNN.

**Comment by jasmainak** [comment]
Could you explain a bit more what you mean by "name is assigned appropriately" ?

**Comment by jasmainak** [comment]
> A similar function would be great for network spiking so that we can generate data files similar to those produced by HNN.

agreed, although I would say that providing access the spikes in clear object oriented way should already make it easy for users to save the spikes easily as they want. I can work on this. Thanks for reporting!

**Comment by rythorpe** [comment]
> Could you explain a bit more what you mean by "name is assigned appropriately" ?

My description was probably a poor choice of words, apologies. I meant to say that the code in `dipole.py` has a small bug where the arg `fname` isn't called in `np.savetxt()`:

```python
def write(self, fname):
    X = np.r_[[self.t, self.dpl['agg'], self.dpl['L2'], self.dpl['L5']]].T
    np.savetxt('dpl2.txt', X, fmt=['%3.3f', '%5.4f', '%5.4f', '%5.4f'],delimiter='\t')
```

**Comment by jasmainak** [comment]
Oh indeed :)

do you want to give a shot at fixing it?

**Comment by rythorpe** [comment]
Sure! I'll take care of this one and add a similar `.write_spikes()` function to the `Network` class if that's cool with you. As you said earlier, all this stuff is simple to access as object attributes, but it'd be nice to have a standard for I/O formatting so that we can potentially implement other batch-oriented visualization tools.

**Comment by jasmainak** [comment]
> I'll take care of this one and add a similar .write_spikes() function to the Network class if that's cool with you

I think this might be more tricky but if you manage to get it working, why not!

**Comment by rythorpe** [comment]
closed by #96
